pkg/operator/ceph/cluster/osd/spec.go

```diff
@@ -619,7 +635,8 @@ func (c *Cluster) makeDeployment(osdProps osdProperties, osd OSDInfo, provisionC
 			InitContainers:     initContainers,
 			Containers: []v1.Container{
 				{
-					Command:         osdStartScript(command, args, c.spec.Storage.FlappingRestartIntervalHours),
+					Command:         append([]string{"bash", "-x", "-c", cephOSDStart, "--"}, command...),
```

Comment:
The issue seemed to be the `--` that I added here. That is making it work right in my environment at least. It tells bash definitively to treat the remaining line as arguments to the script.

Reply:
This works. Thanks @BlaineEXE